src/HDF5.jl

```diff
@@ -1378,26 +1376,41 @@ function getindex(parent::Union{HDF5File, HDF5Group, HDF5Dataset}, r::HDF5Refere
     h5object(obj_id, parent)
 end
 
+# Helper for reading compound types
+function read_row(io::IO, membertype, membersize)
+    row = Any[]
+    for (dtype, dsize) in zip(membertype, membersize)
+        if dtype === String
+            push!(row, strip(String(read(io, UInt8, dsize)),'\0'))
+        elseif dtype <: HDF5.FixedArray
+            push!(row, read(io, eltype(dtype), size(dtype)[1]))
+        else
+            push!(row, read(io, dtype))
+        end
+    end
+    return row
```

Comment:
It would be nice if this whole method could use the set of `read` methods defined immediately above this in the source, but it is not immediately obvious how to do that. At the very least, then, the last block should check that `dtype <: HDF5BitsKind` and then have a final `else` block throw an error.